### PR TITLE
Implemented Rank Functionality

### PR DESCRIPTION
**Pull request by JonathanMathBoi**
Added getRank to puzzle, and added rank printouts to guess and display functions.

### CODE REVIEW

File: src/Puzzle.java [hunk 1]
```diff
@@ -181,6 +195,21 @@ public List<String> getFoundWords() {
...
+    /**
+     * Gets the current rank achived for the puzzle.
+     * 
+     * @return the current rank achived for the puzzle
+     */
+    public Rank getRank() {
+        Rank currentRank = null;
+        for (Rank rank : Rank.values()) {
+            if (earnedPoints >= rank.getRequiredPoints(totalPoints)) {
+                currentRank = rank;
```
Comment (JJSnader) [inline src/Puzzle.java:208]: Breaking the loop after the rank is assigned would save a few iterations.

**Review by JJSnader — APPROVED** [review]
I don't see any problems in this pull.

File: src/Puzzle.java (hunk 1 above)
Comment (JonathanMathBoi) [reply src/Puzzle.java:208]: Will do.

Comment (JonathanMathBoi) [reply src/Puzzle.java:208]: That will only ever return beginner. I could reverse the array but I think that would add more processing than just doing it this way.

Comment (JonathanMathBoi) [reply src/Puzzle.java:208]: But every rank below the current rank also satisfies the condition.

**Pull request merged**